@SameLen annotations that reference private methods fail to parse

Consider the following code (available on [this branch](https://github.com/kelloggm/checker-framework/tree/bad-samelen-inference) as `checker/tests/index/PrivateSameLen.java`):

```
import org.checkerframework.checker.index.qual.SameLen;

public class PrivateSameLen {

  @Pure
  private @SameLen("#1") String getSameLenString(String in) {
    return in;
  }

  private void test() {
    String in = "foo";
    @SameLen("this.getSameLenString(in)") String myStr = getSameLenString(in);
  }
}
```

I expect that this code should be compilable with no warnings. However, I get the following:
```
checker/tests/index/PrivateSameLen.java:12: error: [expression.unparsable] Expression invalid in dependent type annotation: [error for expression: this.getSameLenString(in); error: Invalid 'getSameLenString' because no such method]
    @SameLen("this.getSameLenString(in)") String myStr = getSameLenString(in);
                                          ^
checker/tests/index/PrivateSameLen.java:12: error: [assignment] incompatible types in assignment.
    @SameLen("this.getSameLenString(in)") String myStr = getSameLenString(in);
                                                                         ^
  found   : @SameLen({"in", "this.getSameLenString(in)"}) String
  required: @SameLen("[error for expression: this.getSameLenString(in); error: Invalid 'getSameLenString' because no such method]") String
2 errors
```

Both errors occur because `this.getSameLenString(in)` is apparently unparsable. I don't understand why: it is an in-scope method at this location and it is deterministic, which according to both the documentation in `MethodCall` and in the manual [here](https://checkerframework.org/manual/#java-expressions-as-arguments) are the only requirements.

I think there is another requirement, because if I change `getSameLenString` to `public` rather than `private`, the method compiles with no problems.

Further, the Index Checker is currently inferring this unparsable annotation during local type inference, which causes problems for WPI. The test (on the same branch) `checker/tests/ainfer-index/unannotated/BadSameLen.java` demonstrates this problem in the WPI context.

@smillst @mernst what is the correct spec for expression parsing? In particular, are `private` methods supposed to be allowed in Java Expressions? If not, we should document that and fix the bug in the SameLen Checker related to inference. If so, we should fix whatever is causing these expressions to be unparsable.

--- a/checker/tests/index/PrivateSameLen.java
+++ b/checker/tests/index/PrivateSameLen.java
@@ -1,0 +1,15 @@
+import org.checkerframework.checker.index.qual.SameLen;
+import org.checkerframework.dataflow.qual.Pure;
+
+public class PrivateSameLen {
+
+  @Pure
+  private @SameLen("#1") String getSameLenString(String in) {
+    return in;
+  }
+
+  private void test() {
+    String in = "foo";
+    @SameLen("this.getSameLenString(in)") String myStr = getSameLenString(in);
+  }
+}

--- a/javacutil/src/main/java/org/checkerframework/javacutil/Resolver.java
+++ b/javacutil/src/main/java/org/checkerframework/javacutil/Resolver.java
@@ -367,6 +367,11 @@ public class Resolver {
         if (resolveResult.getKind() == ElementKind.METHOD
             || resolveResult.getKind() == ElementKind.CONSTRUCTOR) {
           methodResult = (ExecutableElement) resolveResult;
+        } else if (resolveResult.getKind() == ElementKind.OTHER
+            && ACCESSERROR.isInstance(resolveResult)) {
+          // Return the inaccessible field that was found
+          methodResult =
+              (ExecutableElement) wrapInvocation(resolveResult, ACCESSERROR_ACCESS, null, null);
         } else {
           methodResult = null;
         }

--- a/javacutil/src/main/java/org/checkerframework/javacutil/Resolver.java
+++ b/javacutil/src/main/java/org/checkerframework/javacutil/Resolver.java
@@ -369,7 +369,7 @@ public class Resolver {
           methodResult = (ExecutableElement) resolveResult;
         } else if (resolveResult.getKind() == ElementKind.OTHER
             && ACCESSERROR.isInstance(resolveResult)) {
-          // Return the inaccessible field that was found
+          // Return the inaccessible method that was found.
           methodResult =
               (ExecutableElement) wrapInvocation(resolveResult, ACCESSERROR_ACCESS, null, null);
         } else {

